taichi/ir/ir.cpp

```diff
@@ -847,6 +841,28 @@ std::string AtomicOpExpression::serialize() {
   }
 }
 
+void AtomicOpExpression::flatten(FlattenContext *ctx) {
+  // replace atomic sub with negative atomic add
+  if (op_type == AtomicOpType::sub) {
+    val.set(Expr::make<UnaryOpExpression>(UnaryOpType::neg, val));
+    op_type = AtomicOpType::add;
+  }
```

Comment:
Why? Could you clarify the reason in the comment? I don't understand.. wasn't `atomic_sub` slimer than `neg+atomic_add`?

Reply:
I didn't make any change here. Please compare this with the old code (`visit(FrontendAtomicStmt*)`)

Reply:
So we should ask @yuanming-hu for reason?

Reply:
I think the reason is simply that the codegen assumes that `sub` is converted to `add` in the previous passes, see https://github.com/taichi-dev/taichi/blob/75335e9576926cd25f6c5e15ed5f3a0265ee91e5/taichi/codegen/codegen_llvm.cpp#L947-L963. It only handles `add`, but not `sub`

Reply:
So let's just add handles for `sub`? It should have no harm but profitable right? Also metal and gl handles sub too, no reason special for llvm.

Reply:
I think the current approach is fine. Adding handling for`sub` will have to cover all the backends, not just for LLVM. Again, let’s always focus one PR for one thing.